plugins/modules/database/misc/redis_set.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Andreas Botzner <andreas at botzner dot com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: redis_set
+short_description: Set key value pairs in Redis
+version_added: 3.7.0
+description:
+   - Set key value pairs in Redis database.
+author: "Andreas Botzner (@paginabianca)"
+options:
+    key:
+        description:
+            - Database key.
+        required: True
+        type: str
+    value:
+        description:
+            - Value that key should be set to.
+        required: False
+        type: str
+    expiration:
+        description:
+            - Expiration time in milliseconds.
+        required: False
+        type: int
+    non_existing:
+        description:
+            - Only set key if it does not already exist.
+        required: False
+        type: bool
+    existing:
+        description:
+            - Only set key if it already exists.
+        required: False
+        type: bool
+    keep_ttl:
+        description:
+            - Retain the time to live associated with the key.
+        required: False
+        type: bool
+    state:
+        description:
+            - State of the key.
+        default: present
+        type: str
+        choices:
+            - present
+            - absent
+
+extends_documentation_fragment:
+  - community.general.redis.documentation
+
+seealso:
+    - module: community.general.redis_data_info
+    - module: community.general.redis
+'''
+
+EXAMPLES = '''
+- name: Set key foo=bar on localhost with no username
+  community.general.redis_set:
+    login_host: localhost
+    login_password: supersecret
+    key: foo
+    value: bar
+    state: present
+
+- name: Set key foo=bar if non existing with expiration of 30s
+  community.general.redis_set:
+    login_host: localhost
+    login_password: supersecret
+    key: foo
+    value: bar
+    non_existing: true
+    expiration: 30000
+    state: present
+
+- name: Set key foo=bar if existing and keep current TTL
+  community.general.redis_set:
+    login_host: localhost
+    login_password: supersecret
+    key: foo
+    value: bar
+    existing: yes
+    keep_ttl: yes
+
+- name: Set key foo=bar on redishost with custom ca-cert file
+  community.general.redis:
+    login_host: redishost
+    login_password: supersecret
+    login_user: someuser
+    validate_certs: yes
+    ssl_ca_certs: /path/to/ca/certs
+    key: foo
+    value: bar
+
+- name: Delete key foo=bar on localhost with no username
```

Comment:
```suggestion
- name: Delete key foo on localhost with no username
```
It will also be deleted if it has another value than bar, right?